src/openloco/windows/build_vehicle.cpp

```diff
@@ -1014,44 +1001,55 @@ namespace openloco::ui::build_vehicle
             }
         }
 
         if (vehicleObj->power != 0)
         {
             if (vehicleObj->mode == TransportMode::rail || vehicleObj->mode == TransportMode::road)
             {
-                _common_format_args[0] = vehicleObj->power;
-                buffer = stringmgr::format_string(buffer, string_ids::stats_power, _common_format_args);
+                FormatArguments args{};
+                args.push(vehicleObj->power);
+                buffer = stringmgr::format_string(buffer, string_ids::stats_power, &args);
             }
         }
 
-        _common_format_args[0] = vehicleObj->weight;
-        buffer = stringmgr::format_string(buffer, string_ids::stats_weight, _common_format_args);
-        _common_format_args[0] = vehicleObj->speed;
-        buffer = stringmgr::format_string(buffer, string_ids::stats_max_speed, _common_format_args);
-
+        {
+            FormatArguments args{};
+            args.push(vehicleObj->weight);
+            buffer = stringmgr::format_string(buffer, string_ids::stats_weight, &args);
+        }
+        {
+            FormatArguments args{};
+            args.push(vehicleObj->speed);
+            buffer = stringmgr::format_string(buffer, string_ids::stats_max_speed, &args);
+        }
         if (vehicleObj->flags & flags_E0::rack_rail)
         {
             auto trackExtraObj = objectmgr::get<track_extra_object>(vehicleObj->rack_rail_type);
-            _common_format_args[0] = vehicleObj->rack_speed;
-            _common_format_args[1] = trackExtraObj->name;
-            buffer = stringmgr::format_string(buffer, string_ids::stats_velocity_on_string, _common_format_args);
+            FormatArguments args{};
+            args.push(vehicleObj->rack_speed);
+            args.push(trackExtraObj->name);
+            buffer = stringmgr::format_string(buffer, string_ids::stats_velocity_on_string, &args);
         }
 
         if (vehicleObj->num_simultaneous_cargo_types != 0)
         {
             {
-                *(reinterpret_cast<uint32_t*>(&_common_format_args[1])) = vehicleObj->max_primary_cargo;
                 auto cargoType = utility::bitscanforward(vehicleObj->primary_cargo_types);
                 if (cargoType != -1)
                 {
                     auto cargoTypes = vehicleObj->primary_cargo_types & ~(1 << cargoType);
-                    auto cargoObj = objectmgr::get<cargo_object>(cargoType);
-                    _common_format_args[0] = vehicleObj->max_primary_cargo == 1 ? cargoObj->unit_name_singular : cargoObj->unit_name_plural;
-                    buffer = stringmgr::format_string(buffer, string_ids::stats_capacity, _common_format_args);
+                    {
+                        auto cargoObj = objectmgr::get<cargo_object>(cargoType);
+                        FormatArguments args{};
+                        args.push(vehicleObj->max_primary_cargo == 1 ? cargoObj->unit_name_singular : cargoObj->unit_name_plural);
```

Comment:
I would prefer a local (temp) var for this string_id. The compiler will optimise it away, but it makes for easier reading.